composer/workflows/bq_copy_across_locations.py

```diff
@@ -0,0 +1,194 @@
+# Set variables
+# --------------------------------------------------------------------------------
+
+# 'master_file_path': This variable will contain the location of the master
+# file.
+master_file_path = Variable.get('master_file_path')
+
+# Source Bucket
+source_bucket = Variable.get('gcs_source_bucket')
+
+# Destination Bucket
+dest_bucket = Variable.get('gcs_dest_bucket')
+
+# --------------------------------------------------------------------------------
+# Set GCP logging
+# --------------------------------------------------------------------------------
+
+# Instantiates a client
+client = google.cloud.logging.Client()
+
+# Connects the logger to the root logging handler; by default this captures
+# all logs at INFO level and higher
+client.setup_logging()
+
+logger = logging.getLogger('bq_copy_us_to_eu_01')
+
+# --------------------------------------------------------------------------------
+# Functions
+# --------------------------------------------------------------------------------
+
+
+def read_master_file(master_file):
+    """
+    Reads the master CSV file that will help in creating Airflow tasks in
+    the DAG dynamically.
+    :param master_file: (String) The file location of the master file,
+                                  e.g. '/home/airflow/framework/master.csv'
+    :return master_record_all: (List) List of Python dictionaries containing
+    the information f
+                                      or a single row in master CSV file.
+    """
+    master_record_all = []
+    logger.info('Reading master_file from : %s' % str(master_file))
+    try:
+        with open(master_file, 'rb') as csv_file:
+            csv_reader = csv.reader(csv_file)
+            next(csv_reader)  # skip the headers
+            for row in csv_reader:
+                logger.info(row)
+                master_record = {
+                    'table_source': row[0],
+                    'table_dest': row[1]
+                }
+                master_record_all.append(master_record)
+            return master_record_all
+    except IOError as e:
+        logger.error('Error opening master_file %s: ' % str(master_file), e)
+
+
+# --------------------------------------------------------------------------------
+# Main DAG
+# --------------------------------------------------------------------------------
+
+# Define a DAG (directed acyclic graph) of tasks.
+# Any task you create within the context manager is automatically added to the
+# DAG object.
+with models.DAG('bq_copy_us_to_eu_01',
+                default_args=default_args,
+                schedule_interval=None) as dag:
+    start = DummyOperator(
+        task_id='start',
+        trigger_rule='all_success'
+    )
+
+    end = DummyOperator(
+        task_id='end',
+
+        trigger_rule='all_success'
+    )
+
+    # Get the table list from master file
+    all_records = read_master_file(master_file_path)
```

Comment:
@fenglu-g Do you have thoughts on doing dynamic calls to construct a DAG a parse time? I thought this was not a recommended practice.